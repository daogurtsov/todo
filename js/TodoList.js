/**
 * Created by dao on 15/01/2017.
 */
import React from 'react'
import Todo from './Todo'
import {List} from 'immutable'

const {func, instanceOf} = React.PropTypes

export class TodoList extends React.Component {

  constructor(props) {
    super(props)
    this.onSubmit = this.onSubmit.bind(this)
    this.render = this.render.bind(this)
    this.toggleClick = this.toggleClick.bind(this)
  }

  onSubmit(event) {
    const {addTodo} = this.props
    const input = event.target
    const text = input.value
    const isEnterKey = (event.which == 13)
    const isLongEnough = text.length > 0

    if (isEnterKey && isLongEnough) {
      input.value = ''
      addTodo(text)
    }
  }

  toggleClick(id) {
    const {toggleTodo} = this.props;
    return event => toggleTodo(id)
  }

  render() {
    return (
      <div className='todo'>
        <input type='text'
          className='todo__entry'
          placeholder='Add todo'
          onKeyDown={this.onSubmit} />
        <ul className='todo__list'>
          {this.props.todos.map(t => (
            <li key={t.get('id')}
              className='todo__item'
              onClick={this.toggleClick(t.get('id'))}>
              <Todo todo={t.toJS()} />
            </li>
          ))}
        </ul>
      </div>
    )
  }
}

TodoList.propTypes = {
  addTodo: func,
  toggleTodo: func,
  todos: instanceOf(List)
}


