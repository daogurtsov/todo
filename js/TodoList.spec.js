/**
 * Created by dao on 18/01/2017.
 */
/* eslint-env node, jest */
import React from 'react'
import {Provider} from 'react-redux'
import  {TodoList as UnwrappedTodoList} from './TodoList'
import TodoList from './containers'
import {createStore} from 'redux'
import reducer from './reducer'
import {addTodo} from './actions'
import {shallow, render} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'
import preload from '../public/data.json'
import {List, Map} from 'immutable'

const todos = List(preload.map(todo => Map(todo)))

const store = createStore(reducer)

test('TodoList snapshot', () => {
  const component = shallow(<UnwrappedTodoList todos={todos} />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})

test('TodoList render all todos', () => {
  const component = render(<UnwrappedTodoList todos={todos} />)
  expect(todos.size).toEqual(component.find('.todo__list li').length)
})

test('Should render with Provider wrapper', () => {
  store.dispatch(addTodo('hehe'))
  store.dispatch(addTodo('hehe2'))
  const component = render(<Provider store={store}><TodoList /></Provider>)
  expect(component.find('.todo__list li').length).toEqual(2)
})
