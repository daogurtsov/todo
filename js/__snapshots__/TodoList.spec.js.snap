exports[`test TodoList snapshot 1`] = `
<div
  className="todo">
  <input
    className="todo__entry"
    onKeyDown={[Function bound onSubmit]}
    placeholder="Add todo"
    type="text" />
  <ul
    className="todo__list">
    <li
      className="todo__item"
      onClick={[Function anonymous]}>
      <Todo
        todo={
          Object {
            "id": "1",
            "isDone": true,
            "text": "hehe"
          }
        } />
    </li>
    <li
      className="todo__item"
      onClick={[Function anonymous]}>
      <Todo
        todo={
          Object {
            "id": "2",
            "isDone": false,
            "text": "hehe1"
          }
        } />
    </li>
  </ul>
</div>
`;
